src/playlist-loader.js

```diff
@@ -354,15 +354,23 @@ const PlaylistLoader = function(srcUrl, hls, withCredentials) {
     if (this.media_) {
       this.trigger('mediachanging');
     }
+
     request = this.hls_.xhr({
-      uri: resolveUrl(loader.master.uri, playlist.uri),
+      uri: playlist.resolvedUri,
       withCredentials
     }, function(error, req) {
       // disposed
       if (!request) {
         return;
       }
 
+      if(request.responseURL) {
```

Comment:
Might be good to add a comment here that we check for the existence of `responseURL` to prevent breaking IE or other browsers that don't support it.